guardian/kindle-gen
# Add ability to run kindle-gen locally

## What does this change?
This PR allows a developer to run the kindle-gen lambda locally, provided they use the correct configuration settings.

## How to test
Follow the instructions in [the README](https://github.com/guardian/kindle-gen/blob/fp/local-kindle-gen/README.md#running-locally) so that you are using the right configuration settings and local environment.

Use `sbt run` to run the lambda, and inspect the CODE S3 bucket to see the uploaded output.

## How can we measure success?
The lambda should execute without error, and upload correctly formatted kindle data to the DEV folder in the CODE S3 bucket.

## Have we considered potential risks?
Pointing to the production bucket could risk the Kindle publishing unintended data, so be sure to follow the instructions in the README closely.

Excessive use of keys used in production (e.g. Accuweather has the same key for PROD and CODE, which I used locally) could cause us to reach the rate limit for that key. Ideally, we would have separate keys for different environments, but so far I have been unable to track down who maintains our relationship with Accuweather (this service is used by many applications in different teams).
